libsql-server/src/namespace/mod.rs

```diff
@@ -348,6 +350,7 @@ struct NamespaceStoreInner<M: MakeNamespace> {
     /// The namespace factory, to create new namespaces.
     make_namespace: M,
     allow_lazy_creation: bool,
+    has_shutdown: RwLock<bool>,
```

Comment:
Why is this a RwLock and not a Mutex? And infact we can probably use AtomicBool instead?

Reply:
> and Ill add to this why is it an async RwLock and not a sync one?

I didn't give much thought to it. The existing code used async lock, since the module had async operations, I used async lock. `has_shutdown`, we can use sync lock/mutex. Should we always default to using sync mutex if that works?

> Why is this a RwLock and not a Mutex? 
The value would be mutated only once, rest all are read operations. So I thought RWLock would be better 

> And infact we can probably use AtomicBool instead?
I guess we can

Reply:
yes, an `AtomicBool` is sufficient